astropy/time/tests/test_basic.py

```diff
@@ -1056,6 +1056,9 @@ def test_to_datetime():
         assert tz.tzname(dt) == tz_dt.tzname()
     assert np.all(time == forced_to_astropy_time)
 
+    with pytest.raises(ValueError):
+        Time('2015-06-30 23:59:60.000').to_datetime()
```

Comment:
Could you test for getting the new error message? This test would pass even without your PR as even then a ValueError was raised.